src/System.Slices/src/System/Span.cs

```diff
@@ -283,45 +288,25 @@ public bool ReferenceEquals(Span<T> other)
 
         public override int GetHashCode()
         {
-            // TODO: Write something better
-            uint hash = unchecked((uint)Length);
-            foreach (T el in this)
+            var hashCode = Length ^ Offset.GetHashCode();
+            if (Object != null)
             {
-                hash = (hash >> 7) | (hash << 25);
-                hash ^= unchecked((uint)el.GetHashCode());
+                hashCode ^= Object.GetHashCode();
             }
-
-            return unchecked((int)(hash));
+            return hashCode;
         }
 
+        /// <summary>
+        /// Checks to see if two spans point at the same memory.  Note that
+        /// this does *not* check to see if the *contents* are equal.
+        /// </summary>
         public bool Equals(Span<T> other)
         {
-            if (ReferenceEquals(other))
-            {
-                return true;
-            }
-
-            if (this.Length != other.Length)
-            {
-                return false;
-            }
-
-            var nonBoxingComparer = EqualityComparer<T>.Default;
-            for (int i = 0; i < Length; i++)
-            {
-                if (!nonBoxingComparer.Equals(
-                    this.GetItemWithoutBoundariesCheck(i), other.GetItemWithoutBoundariesCheck(i)))
-                {
-                    return false;
-                }
-            }
-
-            return true;
+            return ReferenceEquals(other);
```

Comment:
Yeah, I think we should remove (or hide those that cannot be removed) all O(n) comparison instance methods from Span<T>. I think it would be reasonable for these methods to be either static, or on a separate type.